### ISSUE TITLE
Incorrect error from pattern matching a tuple of unions

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#3982]
**Describe the bug**
This is only an error in strict mode.

When using pattern matching (`match` statement) with a tuple - for example `(int | str, int | str)`, pyright will give a warning saying that there is an unhandled case.

**To Reproduce**
Attempt to match against a tuple of unions.

![image](https://user-images.githubusercontent.com/39768508/192042653-677caf14-86fb-4966-b730-02ca6a74b6d8.png)

```python
  /mnt/e/python/test.py:6:11 - error: Cases within match statement do not exhaustively handle all values
    Unhandled type: "tuple[str | int, str | int]"
    If exhaustive handling is not intended, add "case _: pass" (reportMatchNotExhaustive)
  /mnt/e/python/test.py:17:11 - error: Cases within match statement do not exhaustively handle all values
    Unhandled type: "tuple[int | str]"
    If exhaustive handling is not intended, add "case _: pass" (reportMatchNotExhaustive)
```

**Expected behavior**
It should see that all cases are handled and not errored.

**VS Code extension or command-line**
`pyright 1.1.272`

**Comment by erictraut** [comment]
This is a known limitation. We currently have no plans to address this. See https://github.com/microsoft/pylance-release/discussions/3372 for details.
